### PR TITLE
Clean up setup internals

### PR DESCRIPTION
**Pull request by mattseddon**
Related to https://github.com/iterative/vscode-dvc/issues/3434

This PR starts to clean-up the internals of `Setup`. Comments are inline but the main change is removing instances of `DvcReader`, `DvcExecutor`, `GitReader` & `GitExecutor` from `Setup`. All of the required commands should be available through `InternalCommands` which makes things in the class a bit cleaner. 

I also moved command registration out of the class (usual pattern) as then we don't have to worry about commands being "re-registered"/throwing errors when we build `Setup` for testing purposes.

### CODE REVIEW

File: extension/src/extension.ts [hunk 1]
```diff
@@ -131,6 +133,8 @@ export class Extension extends Disposable {
       new InternalCommands(outputChannel, ...clis)
     )
 
+    const status = this.dispose.track(new Status(config, ...clis))
```
Comment (mattseddon) [inline extension/src/extension.ts:136]: [F] There was a bug in the way that we were creating `Status` it did not contain the `dvcViewer`.

File: extension/src/test/suite/setup/util.ts [hunk 2]
```diff
@@ -1,124 +1,60 @@
...
-    mockRoot,
-    mockRunSetup: stub(Runner, 'run').resolves(undefined),
-    mockRunSetupWithRecheck: stub(Runner, 'runWithRecheck').resolves(undefined),
-    mockVersion
-  }
-}
-
 export const buildSetup = (
   disposer: Disposer,
-  dependencies?: {
```
Comment (mattseddon) [inline extension/src/test/suite/setup/util.ts:91]: [F] This was not used anywhere

File: extension/src/setup/index.ts [hunk 3]
```diff
@@ -83,38 +75,23 @@ export class Setup
   private dotFolderWatcher?: Disposer
 
   constructor(
-    stopWatch: StopWatch,
     config: Config,
-    dvcExecutor: DvcExecutor,
-    dvcReader: DvcReader,
-    dvcRunner: DvcRunner,
-    gitExecutor: GitExecutor,
-    gitReader: GitReader,
```
Comment (mattseddon) [inline extension/src/setup/index.ts:92]: [F] All of these `ICli`s sit behind the `InternalCommands` facade. No point to include the functionality twice.

File: extension/src/setup/index.ts [hunk 4]
```diff
@@ -257,10 +241,50 @@ export class Setup
     })
   }
 
+  public async selectFocusedProjects() {
```
Comment (mattseddon) [inline extension/src/setup/index.ts:244]: [F] exposed so it can be registered as a command

File: extension/src/setup/index.ts [hunk 5]
```diff
@@ -310,13 +337,6 @@ export class Setup
     }
   }
 
-  private async initializeDvc() {
```
Comment (mattseddon) [inline extension/src/setup/index.ts:313]: [F] Registered externally and used in that way so we get a "free" telemetry event

File: extension/src/test/cli/util.ts [hunk 6]
```diff
@@ -36,7 +36,7 @@ export const initializeEmptyRepo = async (): Promise<string> => {
     return ''
   }
 
-  await gitExecutor.init(TEMP_DIR)
+  await gitExecutor.gitInit(TEMP_DIR)
```
Comment (mattseddon) [inline extension/src/test/cli/util.ts:39]: [F] (unfortunate but) you can't register the same command name twice in `InternalCommands`

File: extension/src/test/suite/setup/util.ts [hunk 7]
```diff
@@ -1,124 +1,60 @@
...
+import { Status } from '../../../status'
 
 export const TEMP_DIR = join(dvcDemoPath, 'temp-empty-watcher-dir')
 
-const buildSetupDependencies = (
-  disposer: Disposer,
-  mockDvcRoot: string | undefined,
-  mockGitRoot: string | undefined,
-  noGitCommits = true
-) => {
```
Comment (mattseddon) [inline extension/src/test/suite/setup/util.ts:33]: [F] Now that we are using `InternalCommands` we have to build the dependencies in the normal way.

File: extension/src/extension.ts (hunk 1 above)
Comment (mattseddon) [reply extension/src/extension.ts:136]: I'll add a type in a quick follow-up

File: extension/src/setup/index.ts [hunk 8]
```diff
@@ -161,7 +136,16 @@ export class Setup
...
+      if (tryGlobalCli) {
+        return await this.internalCommands.executeCommand(
+          AvailableCommands.GLOBAL_VERSION,
+          cwd
+        )
+      }
+      return await this.internalCommands.executeCommand(
+        AvailableCommands.VERSION,
+        cwd
+      )
```
Comment (mattseddon) [inline extension/src/setup/index.ts:148]: [F] Move the logic out of the "dumb" reader and into here. Wrote some tests to cover it.

**Review by julieg18 — APPROVED** [review]
Great work!

**Review by sroy3 — APPROVED** [review]
Works great and good cleanup job!

**Pull request merged**